src/streams.rs

```diff
@@ -100,7 +113,33 @@ impl BaseStream {
                 }
             },
         };
-        Ok((stream, timeout))
+        Ok(BaseStream::Tls { stream, timeout })
+    }
+
+    #[cfg(feature = "tls-rustls")]
+    fn connect_rustls(host: &str, port: u16, info: &ConnectInfo) -> Result<BaseStream> {
+        let name = DNSNameRef::try_from_ascii_str(host)?;
```

Comment:
It seems like this crate is still pretty low level. Do you think there's value in abstracting tls functionality in a module that supports multiple backends, but offers an identical/abstract interface to use tls functionality?

Reply:
Personally, I did not find it significantly worse than using `native-tls` and there is also the `rustls-connector` crate that provides almost exactly the same API and I did use that in the beginning but the results was not that much simpler.

Considering an additional internal abstraction over `rustls` and `native-tls` more specifically: I think that would only make sense if we decide to avoid exposing the `TlsConnector` or `ClientConfig` for arbitrary TLS setup via our API and provide our own configuration API instead.  Abstracting only the interaction with `TlsStream` and `StreamOwned` does not seem worth it IMHO as `BaseStream` basically already provides exactly this.